guacml/step_tree/model_manager.py

```diff
@@ -0,0 +1,55 @@
+from guacml.step_tree.feature_reducer import FeatureReducer
+from guacml.step_tree.hyper_param_optimizer import HyperParameterOptimizer
+from guacml.step_tree.model_result import ModelResult
+
+from guacml.step_tree.model_runner import ModelRunner
+from .base_step import BaseStep
+
+
+class ModelManager(BaseStep):
+    def __init__(self, model, target, hyper_param_iterations, eval_metric):
+        self.model = model
+        self.target = target
+        self.hyper_param_iterations = hyper_param_iterations
+        self.eval_metric = eval_metric
+        self.splitter = None
+
+    def execute(self, input, metadata):
+        model_runner = ModelRunner(self.model, input, self.target, self.eval_metric, self.splitter)
+        features = self.model.select_features(metadata)
```

Comment:
Did you consider moving the `select_features(metadata)` method from the Model into the ModelManager? It seems to be only used here and it could from the looks of it also be a static method.

We could however go the exactly opposite direction and initialize the Model already with the features and target etc. Then the ModelManager could also get the target and the eval_metric from the Model. Further the model would then already encapsulate all the info it needs to be trained. In addition we could make it immutable and when we want to remove a feature we clone it with the metadata changed for the new instance. This could come in handy when we want to diff models between steps. It would be still able to reference the same dataset as only metadata would have to be changed from one Model revision to another. I would however postpone this bigger refactoring into a separate PR as I'm also still a bit unsure about the downsides of this approach scikit-learn also goes a different direction with their model classes so it might be worth considering staying somewhat consistent with their api choices.